Source/JavaScriptCore/wasm/WasmTypeDefinition.h

```diff
@@ -204,8 +204,108 @@ enum Mutability : uint8_t {
     Immutable = 0
 };
 
-struct FieldType {
-    Type type;
+struct StorageType {
+public:
+    template <typename T>
+    bool is() const { return std::holds_alternative<T>(m_storageType); }
+
+    template <typename T>
+    const T* as() const { ASSERT(is<T>()); return std::get_if<T>(&m_storageType); }
+
+    StorageType() = default;
+
+    explicit StorageType(Type t)
+    {
+        m_storageType = std::variant<Type, PackedType>(t);
+    }
+
+    explicit StorageType(PackedType t)
+    {
+        m_storageType = std::variant<Type, PackedType>(t);
+    }
+
+    // If this is a type, returns itself; if it's a packed type, returns i32
```

Comment:
These kinds of comments aren't very helpful. It would be better to give context about why this is needed, for example explaining that this is the storage type for packed GC types, rather than summarizing what the code does.